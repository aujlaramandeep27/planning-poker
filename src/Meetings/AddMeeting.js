import React from 'react';
import {Link} from 'react-router-dom';

export default function AddMeeting() {
    
    return (
        <div className="uk-flex uk-flex-center">
            <div className="uk-card uk-card-default uk-card-body uk-width-1-2@m">
                <h1 className="uk-text-center uk-heading-divider">Create Meeting</h1>                
                <div className="uk-flex uk-flex-column uk-flex-middle">
                    <input placeholder="Name" className="uk-input uk-margin"></input>
                    <Link to="/meetings/" className="uk-button uk-button-primary">Submit</Link>
                </div>
            </div>
        </div>
    );
}
