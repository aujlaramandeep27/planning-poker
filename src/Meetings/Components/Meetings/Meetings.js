import React, {useEffect} from 'react';
import {connect} from 'react-redux';
import {viewCreateMeeting, viewMeeting} from '../../../Navigation/route-actions';
import {DeleteMeeting, GetMeetings} from '../../Actions/MeetingsActions';
import {Page} from '../../../Common/Page';

export function Meetings({meetings = [], goToMeeting, deleteMeeting, goToCreateMeeting, getMeetings}) {
    useEffect(
        () => {
            getMeetings();
        },
        [getMeetings]
    );

    return (
        <Page title='Meetings'>
            {meetings.map((meeting, index) =>
                <div key={index} className="uk-card uk-card-primary uk-card-body uk-margin-small" onClick={() => goToMeeting(meeting._id)}>
                    {meeting.name}
                    <button
                        className="uk-position-center-right uk-margin-small-right"
                        data-uk-icon="icon: trash"
                        onClick={() => deleteMeeting(meeting._id)}>
                    </button>
                </div>
            )}

            <div className="uk-card uk-card-secondary uk-card-body" onClick={goToCreateMeeting}>
                <h3 className="uk-card-title">+</h3>
            </div>
        </Page>
    );

}

function mapStateToProps(state) {
    return {
        meetings: state.meetings,
        error: state.error
    }
}

function mapDispatchToProps(dispatch) {
    return {
        getMeetings: () => dispatch(GetMeetings()),
        goToMeeting: (meetingId) => dispatch(viewMeeting(meetingId)),
        deleteMeeting: (meetingId) => dispatch(DeleteMeeting(meetingId)),
        goToCreateMeeting: () => dispatch(viewCreateMeeting())
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Meetings)
