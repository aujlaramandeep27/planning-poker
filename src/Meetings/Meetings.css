.meeting-card, .meeting-button {
    background: var(--primary-color);
    background-color: var(--primary-color);
}

.uk-card-title {
    font-family: 'Mandali', sans-serif;
}

.add-card {
    background: var(--secondary-color);
}

.meeting-card:hover, .uk-button-secondary:hover, .add-card:hover {
    background-color: #b1bafc;
}

a:hover {
    text-decoration: none;
}