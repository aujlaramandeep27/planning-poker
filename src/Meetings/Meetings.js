import React from 'react';
import {Link} from 'react-router-dom';

const meetings = ['First', 'Second', 'Third', 'Fourth', 'Fifth'];

export default function Meetings() {
    
    return (
        <div className="uk-flex uk-flex-center">
            <div className="uk-card uk-card-default uk-card-body uk-width-1-2@m">
                <h1 className="uk-text-center uk-heading-divider">Meetings</h1>                
                <div className="uk-flex uk-flex-column uk-flex-middle">
                    { GetButtons() }
                </div>
            </div>
        </div>
    );
}

function GetButtons() {
    if(meetings.length === 0) {
        return <span>No meetings scheduled</span>
    }
    return meetings.map((meeting, index) => <Link to="/" key={index} className="uk-button uk-button-primary uk-width-1-1 uk-margin uk-text-large">{meeting}</Link>);
}