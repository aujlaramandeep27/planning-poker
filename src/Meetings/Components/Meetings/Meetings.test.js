import {Meetings} from "./Meetings";
import {cleanup, render} from "@testing-library/react";
import React from "react";

describe('Container: Meetings', () => {

    afterEach(cleanup);

    it('should render correctly', () => {
        const props = {
            meetings: [{_id: 'id', name: 'test'}],
            error: null,
            goToMeeting: jest.fn(),
            goToCreateMeeting: jest.fn(),
            deleteMeeting: jest.fn(),
            startPollingMeetings: jest.fn(),
            stopPollingMeetings: jest.fn()
        };
        const {container} = render(<Meetings  {...props}/>);
        expect(container).toMatchSnapshot();
    });
});
