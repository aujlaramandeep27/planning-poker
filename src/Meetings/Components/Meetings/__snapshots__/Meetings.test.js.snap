// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meetings renders correctly 1`] = `
<div>
  <div
    class="default-background"
  >
    <div
      class="uk-container uk-text-center@m"
    >
      <h1
        class="uk-heading-divider uk-margin-top"
      >
        Meetings
      </h1>
      <div
        class="uk-align-center uk-width-1-2@m"
      >
        <div
          class="uk-card uk-card-primary uk-card-body uk-margin-small"
        >
          test
          <button
            class="uk-position-center-right uk-margin-small-right"
            data-uk-icon="icon: trash"
          />
        </div>
        <div
          class="uk-card uk-card-secondary uk-card-body"
        >
          <h3
            class="uk-card-title"
          >
            +
          </h3>
        </div>
      </div>
    </div>
  </div>
</div>
`;
